### PR TITLE
modules: Facilitate the use of nrfx drivers

### PR DESCRIPTION
**Pull request by anangl**
This set of commits is intended to facilitate the use of nrfx drivers, and hence nRF peripherals, in Zephyr applications, especially in cases when there are no corresponding Zephyr driver shims.

---

Requires https://github.com/zephyrproject-rtos/hal_nordic/pull/2 to go in first.

### CODE REVIEW

File: modules/Kconfig.nordic [hunk 1]
```diff
@@ -10,121 +10,363 @@ config HAS_NORDIC_DRIVERS
 config HAS_NRFX
 	bool
 
-config NRFX_PRS
-	bool
+if HAS_NRFX
```
Comment (ulfalizer) [inline modules/Kconfig.nordic]: Can put a `depends on` directly on the menu instead, like

```
menu "nrfx drivers"
	depends on HAS_NRFX

...
```

File: modules/Kconfig.nordic [hunk 2]
```diff
@@ -10,121 +10,363 @@ config HAS_NORDIC_DRIVERS
...
-config NRFX_SAADC
-	bool
+config NRFX_RNG
+	bool "Enable RNG driver"
+	depends on HAS_HW_NRF_RNG
 
 config NRFX_RTC
-	bool
+	bool "Enable RTC driver"
+	depends on (HAS_HW_NRF_RTC0 || HAS_HW_NRF_RTC1 || HAS_HW_NRF_RTC2)
```
Comment (ulfalizer) [inline modules/Kconfig.nordic]: Works without the parens too btw:

```
	depends on HAS_HW_NRF_RTC0 || HAS_HW_NRF_RTC1 || HAS_HW_NRF_RTC2
```

File: modules/Kconfig.nordic (hunk 1 above)
Comment (anangl) [reply modules/Kconfig.nordic]: Good to know, I'll update it.

File: modules/Kconfig.nordic (hunk 2 above)
Comment (anangl) [reply modules/Kconfig.nordic]: Ok, I'll remove them.

**Comment by anangl** [comment]
In the hal_nordic module I added a protection against including particular nrfx_config_*.h files directly, not through the common nrfx_config.h. I also moved the change done to the west manifest to a separate commit, for clarity.

**Comment by anangl** [comment]
Updated _west.yml_ with the proper revision of the hal_nordic module as https://github.com/zephyrproject-rtos/hal_nordic/pull/2 has been merged.

**Review by kl-cruz — APPROVED** [review]
I see no problems here. Approved.

**Review by carlescufi — APPROVED** [review]

**Pull request merged**